Modify profiles to get packages building in Conan 2 curated-conan-center-index

datalogics-saharay opened the pull request:
### Changes in this pull request

- Add modifications to profiles to get packages building in the curated-conan-center-index Jenkins pipeline.

**Related to Jira issue**: [APDFL-6103](https://datalogics-jira.atlassian.net/browse/APDFL-6103)


[APDFL-6103]: https://datalogics-jira.atlassian.net/browse/APDFL-6103?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

datalogics-saharay committed:
Add visual-studio-common profile and use it in visual-studio-17

- The visual-studio-17 profile was including the msvc-common profile,
  which had the "compiler.update" setting set. When building the
  Win64 packages in curated-conan-center-index, an error cropped up
  saying that the "settings.compiler.update doesn't exist for Visual Studio".
  As it turns out, the compiler.update setting is only valid for the
  msvc compiler. Let's eliminate the inclusion of the msvc-common profile
  in the visual-studio-17 profile and add a visual-studio-common
  profile instead.

datalogics-robb commented:
The new file needs a newline at the end of the file - it would fail on the Octocat branch for that.

datalogics-robb commented on profiles/visual-studio-17:
```diff
@@ -1,4 +1,4 @@
+include(visual-studio-common)
 [settings]
 os=Windows
 os_build=Windows
```
`os_build` should be an error in Conan 2

datalogics-saharay replied:
Since I'm no longer using the visual studio compiler for CCCI, I'm removing the changes I had made.

datalogics-saharay committed:
Set compiler.threads in AIX profile to get packages to build

- In curated-conan-center-index CI, the AIX packages for ninja and
  b2 were failing to build via "conan create" with the following error:
  "Invalid ID: Invalid: 'settings.compiler.threads' value not defined".

- Let's set the compiler.threads setting in the AIX profile to get
  around this issue.

datalogics-saharay committed:
Set compiler.exception in AIX profile to get packages building

- In curated-conan-center-index CI, the AIX packages for ninja and
  b2 were failing to build via "conan create" with the following error:
  "Invalid ID: Invalid: 'settings.compiler.exception' value not defined".

- The b2 and ninja recipes do not appear to use the compiler.exception
  value, and "None" appears to be a valid value for this setting
  according to the settings.yml file in the .conan2 cache. Let's
  set compiler.exception to "None" in the AIX profile to fix this
  issue.

datalogics-saharay committed:
Add Visual Studio 17 ARM64 profiles

datalogics-saharay committed:
Attempt to format package name for Conan 2

datalogics-saharay committed:
Correct cxxflags configuration so it's recognized by Conan 2

datalogics-saharay committed:
Define os.subsystem for Win64 ARM packages

datalogics-saharay committed:
Define settings.os in common Visual Studio profile

datalogics-saharay committed:
Set compiler.update for Win64 CCCI package build

- The curated-conan-center-index CI was unable to find the
  toolset that is present on the Win64 build node, so it ran
  into the following error:
  [ERROR:vcvars.bat] Toolset directory for version '14.49' was not found.

- To get around this, let's attempt to override the compiler.update
  that is coming from the msvc-common profile.

datalogics-saharay committed:
Force Windows ARM to use 64-bit msys2 package

- The curated-conan-center-index is failing to
  build the automake package for Windows ARM
  with the following error:
  "msys2/cci.latest: Invalid: Only Windows x64 supported".

- Let's modify the profile to force use of the 64-bit
  msys2 package on Windows ARM.

datalogics-saharay committed:
Set compiler.update for msvc 193 profile for CCCI

- When building automake, CI ran into an issue:
  "[ERROR:vcvars.bat] Toolset directory for version '14.39' was not found".

- After poking around on the Win ARM machine, it was discovered that
  the installation of Visual Studio had only 14.38 installed, so let's
  set compiler.update so CI is able to find the toolset directory.

datalogics-saharay committed:
Remove unused visual studio profiles since msvc is preferred with Conan 2
